### PR TITLE
:construction_worker: Run All Steps on MacOS

### PR DESCRIPTION
**Pull request by Paulomart**
**Changes:**

1. Run all build steps on macos slave.

**Issues:**

PR: #996

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -133,6 +134,11 @@ pipeline {
             unstash('post_build_node_modules')
 
             sh('node --version')
+            // We copy the node_modules folder from the slave running
+            // prepare and install steps. That slave may run another OS
+            // then linux. Some dependencies may not be installed if
```
Comment (robpal94) [inline Jenkinsfile]: than

File: Jenkinsfile [hunk 2]
```diff
@@ -133,6 +134,11 @@ pipeline {
             unstash('post_build_node_modules')
 
             sh('node --version')
+            // We copy the node_modules folder from the slave running
```
Comment (FreeYourMinds) [inline Jenkinsfile]: Missing blank line?

File: Jenkinsfile [hunk 3]
```diff
@@ -71,7 +73,6 @@ pipeline {
       }
     }
     // stage('end to end tests') {
```
Comment (FreeYourMinds) [inline Jenkinsfile]: If we don't need it, why keep it. We're in a version control.

File: Jenkinsfile [hunk 4]
```diff
@@ -154,9 +160,10 @@ pipeline {
             unstash('post_build_node_modules')
 
             sh('node --version')
-            // we copy the node_modules folder from the main slave
-            // which runs linux. Some dependencies may not be installed
-            // if they have a os restriction in their package.json
+            // We copy the node_modules folder from the slave running
+            // prepare and install steps. That slave may run another OS
+            // then macos. Some dependencies may not be installed if
+            // they have a os restriction in their package.json.
```
Comment (FreeYourMinds) [inline Jenkinsfile]: .. `an` os restriction...

File: Jenkinsfile (hunk 3 above)
Comment (Paulomart) [reply Jenkinsfile]: Its there to remind us that we need to enable it again... 😭

File: Jenkinsfile (hunk 2 above)
Comment (Paulomart) [reply Jenkinsfile]: Blank line was added.

File: Jenkinsfile [hunk 5]
```diff
@@ -154,9 +160,10 @@ pipeline {
             unstash('post_build_node_modules')
 
             sh('node --version')
-            // we copy the node_modules folder from the main slave
-            // which runs linux. Some dependencies may not be installed
-            // if they have a os restriction in their package.json
+            // We copy the node_modules folder from the slave running
```
Comment (FreeYourMinds) [inline Jenkinsfile]: Missing blank line?

File: Jenkinsfile (hunk 3 above)
Comment (FreeYourMinds) [reply Jenkinsfile]: Suggestion: 

`// ToDo: Readd end-to-end tests`

Clarifies the intention.

**Pull request merged**